tools/run_tests/xds_k8s_test_driver/framework/test_app/base_runner.py

```diff
@@ -35,14 +45,83 @@ class KubernetesBaseRunner:
     def __init__(self,
                  k8s_namespace,
                  namespace_template=None,
-                 reuse_namespace=False):
+                 reuse_namespace=False,
+                 case_name=None,
+                 pod_label_selector=None):
         # Kubernetes namespaced resources manager
         self.k8s_namespace: k8s.KubernetesNamespace = k8s_namespace
         self.reuse_namespace = reuse_namespace
         self.namespace_template = namespace_template or 'namespace.yaml'
+        self.case_name: str = case_name or ""
```

Comment:
> Can we change the case_name variable to be ephemeral?

I'm not sure what you mean by this.

> This will make more sense if a server pod is going to be reused by multiple test cases. E.g., `pod_name`, `pod_name_suffix`.

It was my impression that we didn't reuse pods between test cases. The reason I used the test case name here is because it gives a unique identifier that someone can use to figure out why a particular test case failed.

I suppose the alternative is for each test case that instantiates a deployment to explicitly give it a name. I thought this change might be controversial. I'll do this though if both you and @sergiitk think it is the right move. I might need some input on what to name the existing clients and servers that we spin up in each test.

Reply:
If we move the `case_name` variable to the `run` method, we could reuse the Runner and instantiates the deployment for each case with a more related name. Let's see what @sergiitk think about the naming.

Reply:
I agree with @lidizheng here. Let's make it a `log_name` and move it to `run()`